kafka-client/src/main/java/dev/responsive/kafka/internal/db/CassandraFactTable.java

```diff
@@ -253,44 +304,98 @@ public BoundStatement insert(
       final byte[] value,
       final long epochMillis
   ) {
+    if (ttlResolver.isPresent()) {
+      final Optional<TtlDuration> rowTtl = ttlResolver.get().computeTtl(key, value);
+
+      if (rowTtl.isPresent()) {
+        return insertWithTtl
+            .bind()
+            .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()))
+            .setByteBuffer(DATA_VALUE.bind(), ByteBuffer.wrap(value))
+            .setLong(TIMESTAMP.bind(), epochMillis)
+            .setInt(TTL_SECONDS.bind(), (int) rowTtl.get().toSeconds());
+      }
+    }
+
     return insert
         .bind()
         .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()))
         .setByteBuffer(DATA_VALUE.bind(), ByteBuffer.wrap(value))
-        .setInstant(TIMESTAMP.bind(), Instant.ofEpochMilli(epochMillis));
+        .setLong(TIMESTAMP.bind(), epochMillis);
   }
 
   @Override
-  public byte[] get(final int kafkaPartition, final Bytes key, long minValidTs) {
-    final BoundStatement get = this.get
-        .bind()
-        .setByteBuffer(DATA_KEY.bind(), ByteBuffer.wrap(key.get()))
-        .setInstant(TIMESTAMP.bind(), Instant.ofEpochMilli(minValidTs));
+  public byte[] get(final int kafkaPartition, final Bytes key, long streamTimeMs) {
```

Comment:
the conditional complexity of this method is pretty high -- would be good to split it up into individual methods, perhaps follow the form:

```
if !ttlResolver { simpleGet() }
else if !ttlResolver.needsValueToComputeTtl() { getWithMinTtl() }
else { getWithPostFilterForTtl() }
```